ouroboros-consensus-test/bench-mempool/Main.hs

```diff
@@ -117,20 +150,131 @@ main = do
   Adding TestBlock transactions to a mempool
 -------------------------------------------------------------------------------}
 
-openMempoolWithCapacityFor :: [MempoolCmd TestBlock] ->  IO (MempoolWithMockedLedgerItf IO TestBlock)
-openMempoolWithCapacityFor cmds =
+openMempoolWithCapacityFor ::
+     InitialMempoolAndModelParams IO TestBlock
+  -> [MempoolCmd TestBlock]
+  -> IO (MempoolWithMockedLedgerItf IO TestBlock)
+openMempoolWithCapacityFor params cmds =
     openMempoolWithMockedLedgerItf capacityRequiredByCmds
                                    Tracer.nullTracer
                                    TestBlock.txSize
-                                   TestBlock.sampleMempoolAndModelParams
+                                   params
   where
     capacityRequiredByCmds = Mempool.mkCapacityBytesOverride  totalTxsSize
       where totalTxsSize = sum $ fmap TestBlock.txSize $ txsAddedInCmds cmds
 
-mkNTryAddTxs :: Int -> [MempoolCmd TestBlock.TestBlock]
-mkNTryAddTxs 0 = []
-mkNTryAddTxs n =        [TryAdd [TestBlock.mkTx [] [TestBlock.Token 0]]]
-                <> fmap (TryAdd . (:[]) . mkSimpleTx) (zip [0 .. n - 2] [1 .. n - 1])
-  where
-    mkSimpleTx (x, y) = TestBlock.mkTx [TestBlock.Token (fromIntegral x)]
-                                       [TestBlock.Token (fromIntegral y)]
+{-------------------------------------------------------------------------------
+  Scenarios
```

Comment:
I'm not sure if scenario is the right name for these functions, and whether/how to give them more descriptive names

Reply:
Maybe something like `emptyBackingStoreEmptyChangelogNLinkedTxs`

Reply:
I've written a small DSL instead (credits to @dnadales for the idea), so we can inline the scenarios